src/slack_model.jl

```diff
@@ -71,7 +72,15 @@ function SlackModel(model :: AbstractNLPModel)
     nln=model.meta.nln,
   )
 
-  return SlackModel(meta, model)
+  nls_meta = if isa(model, AbstractNLSModel)
+    NLSMeta(model.nls_meta.nequ,
+            model.meta.nvar + ns,
+            [model.meta.x0; zeros(ns)])
+  else
+    NLSMeta(0, 0)
+  end
```

Comment:
Shouldn't we have two different constructors instead of an `if`?

Reply:
In fact, I'm a bit confused here. Why does SlackModel derive from a more specific type than AbstractNLPModel?

Reply:
Deriving from NLS, Slack is also an NLP, so it doesn't break anything that was working before. I did this to avoid creating a SlackNLSModel duplicating all the code.
But I see that this can be prone to error. Do you think I should create SlackNLSModel?

Reply:
Is it really needed? Why can't we just define `residual`, `jprod_residual`, etc., for `SlackModel`s? If the underlying model doesn't support those methods, there will be an error.

Reply:
Defining the NLS api while Slack was an NLP worked.